packages/server/api/src/app/request-writer/request-writer.module.ts

```diff
@@ -0,0 +1,17 @@
+import { FastifyPluginAsyncTypebox } from '@fastify/type-provider-typebox'
+import { websocketService } from '../websockets/websockets.service'
+import { requestWriterService } from './request-writer.service'
+import { GenerateCodeRequest, GenerateCodeResponse, WebsocketClientEvent, WebsocketServerEvent } from '@activepieces/shared'
+
+export const requestWriterModule: FastifyPluginAsyncTypebox = async () => {
+    websocketService.addListener(WebsocketServerEvent.GENERATE_HTTP_REQUEST, (socket) => {
+        return async (data: GenerateCodeRequest) => {
+            const { prompt } = data
+            const result = await requestWriterService.generateCode({ prompt })
+            const response: GenerateCodeResponse = {
```

Comment:
generateCode should be generateHttpRequest and response type should be GenerateHttpRequestResponse